### DISCUSSION TITLE
CLI / scaffolding

### DISCUSSION THREAD

**Discussion by thedevdavid** [timlrx/tailwind-nextjs-starter-blog#284] [Ideas]
Hi,

As the options are growing for everything (commenting engine, newsletter provider, analytics, etc.) I think a generator would be pretty cool for this project. Example:

```
npx tailwind-nextjs-starter-blog
> choose newsletter provider
> Emailoctopus
> ConvertKit
> Mailchimp

> Now, choose analytics provider
> Fathom
> Plausible
> Google
```

And it'd generate the project based on user answers. What do you think? Is it something that aligns with the project goals?

**Comment by timlrx** [comment]
Most certainly. I would like to reduce the clutter of my blog as well :)

As part of [v2](https://github.com/timlrx/tailwind-nextjs-starter-blog/discussions/275), I am breaking up the newsletter, analytics, comments components into separate components which can be used in the template as required.

The layout templates would also contain a placeholder (currently looking at handlebars or just a simple commented out component), which can be replaced with the user selected components from the CLI input.

**Reply by thedevdavid** [reply]
Oh, I haven't seen that discussion yet. I'll put my comments there instead.
